packages/react-native/android/upload-sourcemaps.gradle

```
@@ -0,0 +1,48 @@
+import groovy.json.JsonBuilder
+import groovy.json.JsonSlurper
+
+
+tasks.register("uploadSourceMapsToBacktrace") {
+    group = "backtrace"
+    description = "This task uploads source maps generated by the react-native builder to Backtrace based on the .backtracejsrc file."
+
+    doLast {
+        // Fetch the build variant (debug or release)
+        // Default to 'release' if not set
+        def buildVariant = project.hasProperty("buildVariant") ? project.buildVariant.toLowerCase() : "release"
+
+        def debugIdPathEnv = System.getenv("DEBUG_ID_PATH")
+
+        def debugIdFile = debugIdPathEnv ? file(debugIdPathEnv) : layout.buildDirectory
+                .dir("intermediates/sourcemaps/react/${buildVariant}/.backtrace-sourcemap-id")
+                .get()
+                .asFile
+
+        if (!debugIdFile.exists()) {
+            println("Backtrace: Cannot find .backtrace-sourcemap-id debug info.")
+            return
+        }
+
+        def sourcemapDestinationDirectory = layout.buildDirectory.dir("generated/sourcemaps/react").get().asFile
+        def mapFiles = fileTree(dir: sourcemapDestinationDirectory, include: '**/*.map').files
+        def mapFile = mapFiles ? mapFiles.iterator().next() : null
+
+        if (!mapFile) {
+            println("Backtrace: Cannot find final source map file.")
```

could you elaborate which file do you mean? e.g. what was supposed to generate it?

---

Two or three source maps are generated during the source map generation process. The last one is the one created by merging two other source maps (one from the metro build and second from hermesc for example). This is what is the final source map. Do you have a better idea for the name?

---

I guess we could be a bit more verbose here. What do you think about this?

```suggestion
            println("Backtrace: Cannot find the final merged sourcemap from Metro and Hermes.")
```